log/config.go

```diff
@@ -0,0 +1,62 @@
+package log
+
+import (
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var encoderConfig = zapcore.EncoderConfig{
+	TimeKey:        "time",
+	LevelKey:       "severity",
+	NameKey:        "logger",
+	CallerKey:      "caller",
+	MessageKey:     "message",
+	StacktraceKey:  "stacktrace",
+	LineEnding:     zapcore.DefaultLineEnding,
+	EncodeLevel:    encodeLevel(),
+	EncodeTime:     zapcore.RFC3339NanoTimeEncoder,
+	EncodeDuration: zapcore.NanosDurationEncoder,
+	EncodeCaller:   zapcore.ShortCallerEncoder,
+}
+
+func encodeLevel() zapcore.LevelEncoder {
+	return func(l zapcore.Level, enc zapcore.PrimitiveArrayEncoder) {
+		switch l {
+		case zapcore.DebugLevel:
+			enc.AppendString("DEBUG")
+		case zapcore.InfoLevel:
+			enc.AppendString("INFO")
+		case zapcore.WarnLevel:
+			enc.AppendString("WARNING")
+		case zapcore.ErrorLevel:
+			enc.AppendString("ERROR")
+		case zapcore.DPanicLevel:
+			enc.AppendString("CRITICAL")
+		case zapcore.PanicLevel:
+			enc.AppendString("ALERT")
+		case zapcore.FatalLevel:
+			enc.AppendString("EMERGENCY")
+		}
+	}
+}
+
+func developmentConfig() *zap.Config {
+	return &zap.Config{
+		Level:            zap.NewAtomicLevelAt(zapcore.DebugLevel),
+		Encoding:         "console",
```

Comment:
Yes and No because `developmentConfig` is using `development` property of zap core (see https://github.com/uber-go/zap/blob/master/config.go#L65) which unlock various extra behavior disabled with production mode.

Reply:
Or maybe we can take another path and get rid of these two configurations strategy for only one configuration ?

For instance, let's say we're using this configuration : 
```go

var encoderConfig = zapcore.EncoderConfig{
	TimeKey:        "time",
	LevelKey:       "severity",
	NameKey:        "logger",
	CallerKey:      "caller",
	MessageKey:     "message",
	StacktraceKey:  "stacktrace",
	LineEnding:     zapcore.DefaultLineEnding,
	EncodeLevel:    encodeLevel(),
	EncodeTime:     zapcore.RFC3339NanoTimeEncoder,
	EncodeDuration: zapcore.NanosDurationEncoder,
	EncodeCaller:   zapcore.ShortCallerEncoder,
}

func defaultConfig() *zap.Config {
	return &zap.Config{
		Level:            zap.NewAtomicLevelAt(zapcore.DebugLevel),
		Encoding:         "json", //required for gcp structured logging
		EncoderConfig:    encoderConfig,
		OutputPaths:      []string{"stdout"},
		ErrorOutputPaths: []string{"stderr"},
	}
}
```

Using this approach, the minimum logging level is `Debug`, everything between `Debug` and `Warning` is written on stdout and >= `error` on stderr.

The stacktrace is automatically added to structured logging from `Error` level.

Does that suit you?